.github/workflows/stack.yml

```diff
@@ -0,0 +1,14 @@
+on: [push]
+name: Stack
+jobs:
+  runhaskell:
+    name: Stack build and test
+    runs-on: ubuntu-latest # or macOS-latest, or windows-latest
```

Comment:
```suggestion
    runs-on: ubuntu-latest # or macOS-latest, or windows-latest
    strategy:
      fail-fast: false
      matrix:
        ghc:
          - "8.10.7"
          - "9.0.2"
          - "9.2.7"
          - "9.4.4"
```

Reply:
for the Stack GitHub Action, I don't think this will work; Stack will pull the GHC version from the resolver in `stack.yaml`.

If it's important to check multiple GHC versions with Stack I can provide more `stack.yaml`s like you suggested already